packages/altair-app/src/app/modules/altair/effects/query-collection.effect.ts

```diff
@@ -155,24 +155,34 @@ export class QueryCollectionEffects {
   }, { dispatch: false });
 
 
-  importCollection$: Observable<Action> = createEffect(() => {
-    return this.actions$
-      .pipe(
-        ofType(collectionActions.IMPORT_COLLECTION),
-        switchMap(() => {
-          return from(openFile({ accept: '.agc' }));
-        }),
-        switchMap((data: string) => this.collectionService.importCollectionDataFromJson(data)),
-        tap(() => this.notifyService.success('Successfully imported collection.')),
-        map(() => new collectionActions.LoadCollectionsAction()),
-        catchError((error) => {
-          const errorMessage = error.message ? error.message : error.toString();
-          this.notifyService.error(`Something went wrong importing collection. Error: ${errorMessage}`);
-          return EMPTY;
-        }),
-      )
+
+  importCollections$: Observable<Action> = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(collectionActions.IMPORT_COLLECTIONS),
+      switchMap(() => {
+        return from(openFiles({ accept: ".agc" }));
+      }),
+      pipe(filter<string[]>((e) => e !== undefined)),
+      switchMap((data: string[]) =>
+        data.map((elem) =>
```

Comment:
`data.forEach` would be more appropriate here, since you're only using the data in the array and not transforming it